demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -10,10 +10,15 @@
 
 #include <utils/common.hpp>
 
+namespace {
+constexpr size_t MAX_NUM_DECODER = 20;
+}
 
-void Cnn::Init(const std::string &model_path, Core & ie, const std::string & deviceName, const cv::Size &new_input_resolution) {
+Cnn::Cnn(const std::string &model_path, Core & ie, const std::string & deviceName, const cv::Size &new_input_resolution) {
     // ---------------------------------------------------------------------------------------------------
-
+    channels_ = 0;
+    time_elapsed_ = 0;
+    ncalls_ = 0;
```

Comment:
`channels_` is set later. You don't need it here. For `time_elapsed_`, `ncalls_` and `end_token`, in/out names in `EncoderDecoderCNN` use member initializer list.

Reply:
Done